plugins/modules/ec2_vpc_vgw.py

```diff
@@ -211,26 +217,26 @@ def wait_for_status(client, module, vpn_gateway_id, status):
                 break
             else:
                 time.sleep(polling_increment_secs)
-        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-            module.fail_json_aws(e, msg="Failure while waiting for status update")
+        except AnsibleEC2Error as e:
```

Comment:
Can't you define a custom waiter instead?

Reply:
added custom waiter https://github.com/ansible-collections/amazon.aws/pull/2331